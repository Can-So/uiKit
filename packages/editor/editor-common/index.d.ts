export * from './utils';
export * from './types';
export * from './contextIdentifier';
export { default as ProviderFactory, WithProviders, Providers, } from './providerFactory';
export * from './styles';
export * from './ui';
