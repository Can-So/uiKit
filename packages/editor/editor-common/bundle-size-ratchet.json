[
  {
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 64416,
      "gzipSize": 16104
    }
  },
  {
    "id": "node_modules.main",
    "name": "node_modules [main]",
    "stats": {
      "size": 212338,
      "gzipSize": 62562
    }
  },
  {
    "id": "node_modules.async",
    "name": "node_modules [async]",
    "stats": {
      "size": 16063,
      "gzipSize": 6383
    }
  },
  {
    "id": "atlaskit.core.main",
    "name": "main",
    "stats": {
      "size": 102495,
      "gzipSize": 26415
    }
  },
  {
    "id": "atlaskit.editor.main",
    "name": "main",
    "stats": {
      "size": 44915,
      "gzipSize": 10570
    }
  },
  {
    "id": "atlaskit.elements.main",
    "name": "main",
    "stats": {
      "size": 19557,
      "gzipSize": 5632
    }
  },
  {
    "id": "atlaskit.elements.async",
    "name": "async",
    "stats": {
      "size": 7738,
      "gzipSize": 2890
    }
  },
  {
    "id": "atlaskit.people-and-teams.main",
    "name": "main",
    "stats": {
      "size": 68853,
      "gzipSize": 12328
    }
  },
  {
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 502268,
      "gzipSize": 130935
    }
  },
  {
    "id": "combined.async",
    "name": "async",
    "stats": {
      "size": 23737,
      "gzipSize": 9031
    }
  }
]