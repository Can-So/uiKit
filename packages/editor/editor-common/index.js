export * from './utils';
export { default as ProviderFactory, WithProviders, } from './providerFactory';
export * from './styles';
export * from './ui';
//# sourceMappingURL=index.js.map