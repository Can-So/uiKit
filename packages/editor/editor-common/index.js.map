{"version":3,"file":"index.js","sourceRoot":"","sources":["src/index.ts"],"names":[],"mappings":"AAAA,cAAc,SAAS,CAAC;AAKxB,OAAO,EACL,OAAO,IAAI,eAAe,EAC1B,aAAa,GAEd,MAAM,mBAAmB,CAAC;AAE3B,cAAc,UAAU,CAAC;AACzB,cAAc,MAAM,CAAC"}