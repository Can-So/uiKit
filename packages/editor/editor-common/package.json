{
  "name": "@atlaskit/editor-common",
  "version": "9.4.0",
  "description": "A package that contains common classes and components for editor and renderer",
  "license": "Apache-2.0",
  "main": "dist/es5/index.js",
  "atlaskit:src": "src/index.ts",
  "jsnext:main": "dist/es2015/index.js",
  "module": "dist/es2015/index.js",
  "types": "dist/es5/index.d.ts",
  "sideEffects": false,
  "scripts": {
    "build:json-schema": "mkdir -p dist && cp -r json-schema dist",
    "generate:json-schema": "../../../node_modules/.bin/tsc -p build/json-schema && node ./dist/json-schema-cli/index.js ./src/schema/nodes/doc.ts > ./json-schema/v1/full.json"
  },
  "files": [
    "dist"
  ],
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Oscar W",
    "Stanislav S"
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/emoji": "^34.2.0",
    "@atlaskit/icon": "^11.3.0",
    "@atlaskit/media-card": "^27.1.4",
    "@atlaskit/mention": "^11.1.4",
    "@atlaskit/profilecard": "^3.7.1",
    "@atlaskit/util-shared-styles": "^2.10.3",
    "@types/prosemirror-model": "^1.0.0",
    "@types/prosemirror-view": "^1.0.0",
    "ajv": "^4.11.3",
    "classnames": "^2.2.5",
    "css-color-names": "0.0.4",
    "date-fns": "^1.29.0",
    "linkify-it": "^2.0.3",
    "prosemirror-model": "^1.2.0",
    "prosemirror-view": "^1.3.0",
    "raf-schd": "^2.1.0",
    "styled-components": "1.4.6 - 3",
    "typescript": "~2.6.2"
  },
  "peerDependencies": {
    "@atlaskit/media-core": "^18.0.3",
    "react": "^16.3.0",
    "react-dom": "^16.3.0"
  },
  "devDependencies": {
    "@atlaskit/editor-json-transformer": "^3.0.6",
    "@atlaskit/editor-test-helpers": "^3.1.8",
    "@atlaskit/media-core": "^18.0.3",
    "@atlaskit/webdriver-runner": "^0.0.3",
    "@types/chai": "^4.0.4",
    "chai": "^4.1.2",
    "json-schema-diff-validator": "^0.0.5",
    "sinon": "^2.2.0"
  }
}
