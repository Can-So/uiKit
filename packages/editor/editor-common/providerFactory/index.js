export { WithProviders } from './withProviders';
var ProviderFactory = /** @class */ (function () {
    function ProviderFactory() {
        this.providers = new Map();
        this.subscribers = new Map();
    }
    ProviderFactory.create = function (providers) {
        var providerFactory = new ProviderFactory();
        Object.keys(providers).forEach(function (name) {
            providerFactory.setProvider(name, providers[name]);
        });
        return providerFactory;
    };
    ProviderFactory.prototype.destroy = function () {
        this.providers.clear();
        this.subscribers.clear();
    };
    ProviderFactory.prototype.isEmpty = function () {
        return !this.providers.size && !this.subscribers.size;
    };
    ProviderFactory.prototype.setProvider = function (name, provider) {
        // Do not trigger notifyUpdate if provider is the same.
        if (this.providers.get(name) === provider) {
            return;
        }
        if (provider) {
            this.providers.set(name, provider);
        }
        else {
            this.providers.delete(name);
        }
        this.notifyUpdated(name, provider);
    };
    ProviderFactory.prototype.removeProvider = function (name) {
        this.providers.delete(name);
        this.notifyUpdated(name);
    };
    ProviderFactory.prototype.subscribe = function (name, handler) {
        var handlers = this.subscribers.get(name) || [];
        handlers.push(handler);
        this.subscribers.set(name, handlers);
        var provider = this.providers.get(name);
        if (provider) {
            handler(name, provider);
        }
    };
    ProviderFactory.prototype.unsubscribe = function (name, handler) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        var index = handlers.indexOf(handler);
        if (index !== -1) {
            handlers.splice(index, 1);
        }
        if (handlers.length === 0) {
            this.subscribers.delete(name);
        }
        else {
            this.subscribers.set(name, handlers);
        }
    };
    ProviderFactory.prototype.unsubscribeAll = function (name) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        this.subscribers.delete(name);
    };
    ProviderFactory.prototype.hasProvider = function (name) {
        return this.providers.has(name);
    };
    ProviderFactory.prototype.notifyUpdated = function (name, provider) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        handlers.forEach(function (handler) {
            handler(name, provider);
        });
    };
    return ProviderFactory;
}());
export default ProviderFactory;
//# sourceMappingURL=index.js.map