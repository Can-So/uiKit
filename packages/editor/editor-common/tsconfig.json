{
  "extends": "../../../tsconfig.editor.json",
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "./docs/**/*.ts",
    "./docs/**/*.tsx",
    "./tests/**/*.ts",
    "./tests/**/*.tsx",
    "./__tests__/**/*.ts",
    "./__tests__/**/*.tsx",
    "test-helpers/**/*.ts",
    "test-helpers/**/*.tsx"
  ],
  "exclude": [
    "node_modules",
    "src/json-schema"
  ]
}
