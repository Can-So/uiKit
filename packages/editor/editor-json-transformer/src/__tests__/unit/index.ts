import {
  createEditor,
  doc,
  // Node
  blockquote,
  ul,
  ol,
  li,
  code_block,
  emoji,
  br,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  mediaGroup,
  media,
  mention,
  panel,
  panelNote,
  p,
  hr,
  table,
  th,
  tr,
  td,
  // Marks
  code,
  em,
  a,
  strike,
  strong,
  subsup,
  textColor,
  underline,
} from '@atlaskit/editor-test-helpers';
import { ProviderFactory } from '@atlaskit/editor-common';
import { emoji as emojiData } from '@atlaskit/util-data-test';

import { JSONTransformer } from '../../index';
import emojiPlugin from '../../../../editor-core/src/plugins/emoji';
import mentionsPlugin from '../../../../editor-core/src/plugins/mentions';
import codeBlockPlugin from '../../../../editor-core/src/plugins/code-block';
import mediaPlugin from '../../../../editor-core/src/plugins/media';
import textColorPlugin from '../../../../editor-core/src/plugins/text-color';
import panelPlugin from '../../../../editor-core/src/plugins/panel';
import listPlugin from '../../../../editor-core/src/plugins/lists';
import rulePlugin from '../../../../editor-core/src/plugins/rule';
import tablesPlugin from '../../../../editor-core/src/plugins/table';

const transformer = new JSONTransformer();
const toJSON = node => transformer.encode(node);
const parseJSON = node => transformer.parse(node);
const emojiProvider = emojiData.testData.getEmojiResourcePromise();

describe('JSONTransformer:', () => {
  describe('encode', () => {
    const editor = (doc: any) =>
      createEditor({
        doc,
        editorPlugins: [
          emojiPlugin,
          mentionsPlugin,
          codeBlockPlugin(),
          mediaPlugin(),
          textColorPlugin,
          panelPlugin,
          listPlugin,
          rulePlugin,
          tablesPlugin(),
        ],
        providerFactory: ProviderFactory.create({ emojiProvider }),
      });

    it('should serialize common nodes/marks as ProseMirror does', () => {
      const { editorView } = editor(
        doc(
          p(
            strong('>'),
            ' Atlassian: ',
            br(),
            a({ href: 'https://atlassian.com' })('Atlassian'),
          ),
          p(
            em('hello'),
            underline('world'),
            code('!'),
            subsup({ type: 'sub' })('sub'),
            'plain text',
            strike('hey'),
            textColor({ color: 'red' })('Red :D'),
          ),
          ul(li(p('ichi')), li(p('ni')), li(p('san'))),
          ol(li(p('ek')), li(p('dui')), li(p('tin'))),
          blockquote(p('1')),
          h1('H1'),
          h2('H2'),
          h3('H3'),
          h4('H4'),
          h5('H5'),
          h6('H6'),
          p(emoji({ shortName: ':joy:' })()),
          panel()(p('hello from panel')),
          panelNote(p('hello from note panel')),
          hr(),
        ),
      );
      const pmDoc = editorView.state.doc;
      expect(toJSON(pmDoc)).toMatchSnapshot();
    });

    it('should strip optional attrs from media node', () => {
      const { editorView } = editor(
        doc(
          mediaGroup(
            media({
              id: 'foo',
              type: 'file',
              collection: '',
              __fileName: 'foo.png',
              __displayType: 'thumbnail',
              __fileMimeType: 'image/png',
              __fileSize: 1234,
            })(),
          ),
        ),
      );
      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'mediaGroup',
            content: [
              {
                type: 'media',
                attrs: {
                  id: 'foo',
                  type: 'file',
                  collection: '',
                },
              },
            ],
          },
        ],
      });
    });

    it('should strip unused optional attrs from mention node', () => {
      const { editorView } = editor(
        doc(
          p(
            mention({
              id: 'id-rick',
              text: '@Rick Sanchez',
            })(),
          ),
        ),
      );

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'mention',
                attrs: {
                  id: 'id-rick',
                  text: '@Rick Sanchez',
                  accessLevel: '',
                },
              },
            ],
          },
        ],
      });
    });

    it('should not strip accessLevel from mention node', () => {
      const { editorView } = editor(
        doc(
          p(
            mention({
              accessLevel: 'CONTAINER',
              id: 'foo',
              text: 'fallback',
              userType: 'APP',
            })(),
          ),
        ),
      );

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'mention',
                attrs: {
                  id: 'foo',
                  text: 'fallback',
                  userType: 'APP',
                  accessLevel: 'CONTAINER',
                },
              },
            ],
          },
        ],
      });
    });

    it('should strip uniqueId from codeBlock node', () => {
      const { editorView } = editor(
        doc(
          code_block({
            language: 'javascript',
            uniqueId: 'foo',
          })('var foo = 2;'),
        ),
      );

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'codeBlock',
            attrs: {
              language: 'javascript',
            },
            content: [
              {
                type: 'text',
                text: 'var foo = 2;',
              },
            ],
          },
        ],
      });
    });

    it('should strip optional attributes from link mark', () => {
      const { editorView } = editor(
        doc(p(a({ href: 'https://atlassian.com' })('Atlassian'))),
      );

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'Atlassian',
                marks: [
                  {
                    type: 'link',
                    attrs: {
                      href: 'https://atlassian.com',
                    },
                  },
                ],
              },
            ],
          },
        ],
      });
    });

    it('should preserve optional attributes if they are !== null', () => {
      const { editorView } = editor(
        doc(
          p(
            a({
              href: 'https://atlassian.com',
              __confluenceMetadata: { linkType: '' },
            })('Atlassian'),
          ),
        ),
      );

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'Atlassian',
                marks: [
                  {
                    type: 'link',
                    attrs: {
                      href: 'https://atlassian.com',
                      __confluenceMetadata: { linkType: '' },
                    },
                  },
                ],
              },
            ],
          },
        ],
      });
    });

    it('should strip language=null from codeBlock node', () => {
      const { editorView } = editor(doc(code_block()('var foo = 2;')));

      expect(toJSON(editorView.state.doc)).toEqual({
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'codeBlock',
            attrs: {},
            content: [
              {
                type: 'text',
                text: 'var foo = 2;',
              },
            ],
          },
        ],
      });
    });

    [
      { nodeName: 'tableCell', schemaBuilder: td },
      { nodeName: 'tableHeader', schemaBuilder: th },
    ].forEach(({ nodeName, schemaBuilder }) => {
      it(`should strip unused optional attrs from ${nodeName} node`, () => {
        const { editorView } = editor(
          doc(table()(tr(schemaBuilder({ colspan: 2 })(p('foo'))))),
        );

        expect(toJSON(editorView.state.doc)).toEqual({
          version: 1,
          type: 'doc',
          content: [
            {
              type: 'table',
              attrs: {
                isNumberColumnEnabled: false,
                layout: 'default',
              },
              content: [
                {
                  type: 'tableRow',
                  content: [
                    {
                      type: nodeName,
                      attrs: {
                        colspan: 2,
                      },
                      content: [
                        {
                          type: 'paragraph',
                          content: [
                            {
                              type: 'text',
                              text: 'foo',
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        });
      });
      [[0], [200], [100, 200], [100, 0]].forEach(colwidth => {
        describe(`when colwidth=${JSON.stringify(colwidth)}`, () => {
          it(`should preserve colwidth attributes as an array of widths`, () => {
            const { editorView } = editor(
              doc(table()(tr(schemaBuilder({ colwidth })(p('foo'))))),
            );
            expect(toJSON(editorView.state.doc)).toEqual({
              version: 1,
              type: 'doc',
              content: [
                {
                  type: 'table',
                  attrs: {
                    isNumberColumnEnabled: false,
                    layout: 'default',
                  },
                  content: [
                    {
                      type: 'tableRow',
                      content: [
                        {
                          type: nodeName,
                          attrs: {
                            colwidth,
                          },
                          content: [
                            {
                              type: 'paragraph',
                              content: [
                                {
                                  type: 'text',
                                  text: 'foo',
                                },
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            });
          });
        });
      });
    });
  });

  describe('parse', () => {
    it('should convert ADF to PM representation', () => {
      const adf = {
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'hello',
              },
            ],
          },
        ],
      };
      expect(parseJSON(adf)).toEqualDocument(doc(p('hello')));
    });
  });

  it('should throw an error if not ADF-like', () => {
    const badADF = {
      type: 'paragraph',
      content: [{ type: 'text', content: 'hello' }],
    };
    expect(() => parseJSON(badADF)).toThrowError(
      'Expected content format to be ADF',
    );
  });

  it('should throw an error if not a valid PM document', () => {
    const badADF = {
      type: 'doc',
      content: [{ type: 'fakeNode', content: 'hello' }],
    };
    expect(() => parseJSON(badADF)).toThrowError(
      /Invalid input for Fragment.fromJSON/,
    );
  });
});
