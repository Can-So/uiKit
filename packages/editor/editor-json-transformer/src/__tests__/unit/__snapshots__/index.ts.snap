// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONTransformer: encode should serialize common nodes/marks as ProseMirror does 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": ">",
          "type": "text",
        },
        Object {
          "text": " Atlassian: ",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "href": "https://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Atlassian",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "hello",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "underline",
            },
          ],
          "text": "world",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "!",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "type": "sub",
              },
              "type": "subsup",
            },
          ],
          "text": "sub",
          "type": "text",
        },
        Object {
          "text": "plain text",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strike",
            },
          ],
          "text": "hey",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "color": "red",
              },
              "type": "textColor",
            },
          ],
          "text": "Red :D",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "ichi",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "ni",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "san",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "ek",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "dui",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "tin",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "1",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 2,
      },
      "content": Array [
        Object {
          "text": "H2",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 3,
      },
      "content": Array [
        Object {
          "text": "H3",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 4,
      },
      "content": Array [
        Object {
          "text": "H4",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 5,
      },
      "content": Array [
        Object {
          "text": "H5",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 6,
      },
      "content": Array [
        Object {
          "text": "H6",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "id": "",
            "shortName": ":joy:",
            "text": "",
          },
          "type": "emoji",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "hello from panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
    Object {
      "attrs": Object {
        "panelType": "note",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "hello from note panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
    Object {
      "type": "rule",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
